### PR TITLE
 Ask users to leave a review for Matomo

### LINKED ISSUE #14086: Ask users to leave a review for Matomo
* We already have a review feature in Matomo with the thumbs up/down next to certain features. The feedback goes to us and we always read them and try to improve things based on it and listen in general. When there was a thumbs up, in the sentence that says "thank you..." it would be good to also motivate users to leave a review on a feedback site like Capterra, TrustRadius, G2Crowd, ...
* On the help page we could add a section "Like Matomo? Please give us a review... " and link to the same sites...
* Maybe after - for example - 50 logins, we could also show a popup (or 90 days whichever happens first) asking to leave a review with choice [Remind me later, Don't remind be again] . If remind me again is selected, we would maybe remind again in 90 days? or earlier? Or maybe there are better ways without being annoying? Also instead of a popup it could be a nice looking full page and a "next/continue" button... 

This should be added to the Feedback plugin.

### PR DESCRIPTION
**Pull request by katebutler**
Fixes #14086

### CODE REVIEW

**Comment by katebutler** [comment]
Sample screenshot below. @Findus23 Thomas has suggested that you may be able to help with the logos for the review sites - the Product Hunt one is OK as they provide official branding resources, but the others I've just taken off the internet. They need transparent backgrounds and we may need to play around with the size a bit as well, particularly for SaasWorthy.

**Review by Findus23 — CHANGES_REQUESTED** [review]
Hi @katebutler,

Could you please update the images with the SVGs I sent you?
Thanks

**Comment by mattab** [comment]
@katebutler can you check @Findus23 comment from 13 days ago?

**Comment by katebutler** [comment]
Have replaced the icons with the SVGs. Has caused quite a few UI tests to fail because of the popup showing up over the top of the expected dashboard. My own test that the popup doesn't show up when it's not expected is also failing so hopefully once I fix that it will sort the others out.

**Comment by katebutler** [comment]
Ready for review/merge

File: plugins/Feedback/Controller.php [hunk 1]
```diff
@@ -23,4 +27,19 @@ function index()
...
     }
+
+    /**
+     * Store the next date that the feedback reminder popup should be displayed to this user.
+     */
+    public function updateFeedbackReminderDate()
+    {
+        $nextReminder = Common::getRequestVar('nextReminder');
+        // -1 means "never remind me again", otherwise add the interval onto today's date
+        if ((int)$nextReminder !== -1) {
```
Comment (mattab) [inline plugins/Feedback/Controller.php]: maybe could use a const named appropriately and remove the comment?

File: plugins/Feedback/angularjs/feedback-popup/feedback-popup.directive.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    >
+
+        <div class="intro">
+            <h2>{{'Feedback_ReviewMatomoTitle'|translate }}</h2>
+        </div>
+
+        <div ng-include="'plugins/Feedback/angularjs/feedback-popup/review-links.directive.html'"></div>
+
+        <div class="footer">
+            <input type="button" value="Remind me later" role="yes"/>
```
Comment (mattab) [inline plugins/Feedback/angularjs/feedback-popup/feedback-popup.directive.html]: this string is not translated yet (line below as well)

File: plugins/Feedback/Controller.php [hunk 3]
```diff
@@ -23,4 +27,19 @@ function index()
         $view->piwikVersion = Version::VERSION;
         return $view->render();
     }
+
+    /**
+     * Store the next date that the feedback reminder popup should be displayed to this user.
+     */
+    public function updateFeedbackReminderDate()
+    {
+        $nextReminder = Common::getRequestVar('nextReminder');
```
Comment (mattab) [inline plugins/Feedback/Controller.php]: maybe add auth check, that user is not anonymous?

**Comment by mattab** [comment]
@katebutler noticed tests are failing with:

```
There were 3 errors:
1) Piwik\Plugins\Feedback\tests\Unit\FeedbackTest::test_shouldPromptForFeedback_nextReminderDateInPast
Exception: General_ExceptionInvalidDateBeforeFirstWebsite: 2019
/home/travis/build/matomo-org/matomo/core/Date.php:161
/home/travis/build/matomo-org/matomo/plugins/Feedback/Feedback.php:96
/home/travis/build/matomo-org/matomo/plugins/Feedback/tests/Integration/FeedbackTest.php:100
2) Piwik\Plugins\Feedback\tests\Unit\FeedbackTest::test_shouldPromptForFeedack_nextReminderDateToday
Exception: General_ExceptionInvalidDateBeforeFirstWebsite: 2019
/home/travis/build/matomo-org/matomo/core/Date.php:161
/home/travis/build/matomo-org/matomo/plugins/Feedback/Feedback.php:96
/home/travis/build/matomo-org/matomo/plugins/Feedback/tests/Integration/FeedbackTest.php:108
3) Piwik\Plugins\Feedback\tests\Unit\FeedbackTest::test_shouldPromptForFeedack_nextReminderDateInFuture
Exception: General_ExceptionInvalidDateBeforeFirstWebsite: 2019
/home/travis/build/matomo-org/matomo/core/Date.php:161
/home/travis/build/matomo-org/matomo/plugins/Feedback/Feedback.php:96
/home/travis/build/matomo-org/matomo/plugins/Feedback/tests/Integration/FeedbackTest.php:116

```

**Comment by mattab** [comment]
PR looks good otherwise, looking forward to merging this :+1:

**Comment by katebutler** [comment]
`FeedbackTest` is now fixed.

**Pull request merged**